### PR TITLE
Migrate plank to use the new k8s clients

### PR DESCRIPTION
**Pull request by stevekuznetsov**
We only ever update the ProwJobStatus so I changed the API call we make to reflect that.

Signed-off-by: Steve Kuznetsov skuznets@redhat.com

Depends on #11060

/assign @cjwagner @fejta @munnerz
/cc @sebastienvas @BenTheElder @krzyzacy

### CODE REVIEW

**Comment by krzyzacy** [comment]
pass tests, same for other PRs :-)

File: prow/plank/controller.go [hunk 1]
```diff
@@ -186,26 +188,25 @@ func (c *Controller) Sync() error {
 	// TODO: Replace the following filtering with a field selector once CRDs support field selectors.
 	// https://github.com/kubernetes/kubernetes/issues/53459
 	var k8sJobs []prowapi.ProwJob
-	for _, pj := range pjs {
+	for _, pj := range pjs.Items {
 		if pj.Spec.Agent == prowapi.KubernetesAgent {
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply prow/plank/controller.go]: I didnt change this!

[the remark this answers is not in the document]
Comment (fejta) [reply prow/plank/controller.go]: it should change regardless (I gave an lgtm, you don't need to address it)

**Comment by stevekuznetsov** [comment]
@BenTheElder @krzyzacy @munnerz so @cjwagner is out for a week -- do y'all feel comfortable reviewing this or should we wait on him?

**Comment by krzyzacy** [comment]
There's no functional changes, lgtm from me.

And also given the switch to most of other controllers have merged already, let's get this in as well :-) cc @Katharine

**Pull request merged**